src/types.jl

```diff
@@ -1,259 +1,276 @@
+    RI,
+    UpdateTs,
+    UpdateQs,
+    U,
+}
+    Tin::FT
+    qin::FT
+    ρin::FT
+    Ts_guess::FT
+    qs_guess::FT
+    Φs::FT
+    Δz::FT
+    d::FT
+    u_int::U
+    u_sfc::U
     roughness_model::RM
+    gustiness_model::GM
+    roughness_inputs::RI
+    update_Ts!::UpdateTs
+    update_qs!::UpdateQs
+    shf::Union{Nothing, FT}
+    lhf::Union{Nothing, FT}
+    ustar::Union{Nothing, FT}
+    Cd::Union{Nothing, FT}
+    Ch::Union{Nothing, FT}
 end
 
-function Coefficients(
-    state_in::SVA,
-    state_sfc::SVB,
-    Cd::FT,
-    Ch::FT;
-    gustiness::FT = FT(1),
-    beta::FT = FT(1),
-    roughness_model::RM = ScalarRoughness(),
-) where {SVA, SVB, FT, RM}
-    return Coefficients{FT, SVA, SVB, RM}(
-        state_in,
-        state_sfc,
-        Cd,
-        Ch,
-        gustiness,
-        beta,
+function SurfaceFluxInputs(
+    Tin::FT,
+    qin::FT,
+    ρin::FT,
+    Ts_guess::FT,
+    qs_guess::FT,
+    Φs::FT,
+    Δz::FT,
+    d::FT,
+    u_int,
+    u_sfc,
+    roughness_model::RM,
+    gustiness_model::GM,
+    roughness_inputs::RI,
+    update_Ts!,
+    update_qs!,
+    flux_specs::FluxSpecs{FT},
+) where {FT, RM <: AbstractRoughnessSpec, GM <: AbstractGustinessSpec, RI}
+    u_int_tuple = _normalize_velocity(u_int, FT)
+    u_sfc_tuple = _normalize_velocity(u_sfc, FT)
+    return SurfaceFluxInputs{
+        FT,
+        RM,
+        GM,
+        RI,
+        typeof(update_Ts!),
+        typeof(update_qs!),
+        typeof(u_int_tuple),
+    }(
+        Tin,
+        qin,
+        ρin,
+        Ts_guess,
+        qs_guess,
+        Φs,
+        Δz,
+        d,
+        u_int_tuple,
+        u_sfc_tuple,
         roughness_model,
+        gustiness_model,
+        roughness_inputs,
+        update_Ts!,
+        update_qs!,
+        flux_specs.shf,
+        flux_specs.lhf,
+        flux_specs.ustar,
+        flux_specs.Cd,
+        flux_specs.Ch,
     )
 end
 
+@inline function _normalize_velocity(u::NTuple{2, T}, ::Type{FT}) where {T, FT}
+    return (convert(FT, u[1]), convert(FT, u[2]))
+end
+function _normalize_velocity(u::AbstractVector, ::Type{FT}) where {FT}
+    length(u) == 2 ||
+        throw(ArgumentError("Velocity vectors must have two horizontal components."))
+    return (convert(FT, u[1]), convert(FT, u[2]))
+end
+_normalize_velocity(u::Nothing, ::Type{FT}) where {FT} = (zero(FT), zero(FT))
+
+Base.@kwdef mutable struct SurfaceFluxIterationState{FT}
+    Ts::FT = FT(0)
+    qs::FT = FT(0)
+    gustiness::FT = FT(1)
+    ustar::FT = FT(0.1)
+    L_MO::FT = FT(10)
+    shf::FT = FT(0)
+    lhf::FT = FT(0)
+    Cd::FT = FT(0)
+    Ch::FT = FT(0)
+    evaporation::FT = FT(0)
+    ρ_sfc::FT = FT(1)
+    buoyancy_flux::FT = FT(0)
+end
 
-"""
-    ValuesOnly
-
-Input container, given only surface state variables, roughness lengths,
-initial obukhov length and gustiness.
+struct SimilarityScales{FT}
+    u_star::FT
+    dsev_star::FT
+    q_star::FT
+    L_star::FT
+    theta_v_star::FT
+    ell_u::FT
```

Comment:
`\scrl` or `ell` here was inspired somewhat by ClimaOcean nomenclature, `z0_u ,z0_q, z0_theta` as in the profile recovery functions may be a clearer description